surfsense_backend/app/routes/search_source_connectors_routes.py

```
@@ -57,6 +62,33 @@ async def list_github_repositories(
+        # Initialize GoogleDriveConnector with the provided tokens
+        drive_client = GoogleDriveConnector(
+            access_token=tokens_request.access_token,
+            refresh_token=tokens_request.refresh_token
+        )
+        # Fetch files from Google Drive
+        files = drive_client.list_files()
+        return files
+    except ValueError as e:
+        # Handle invalid token error specifically
+        logger.error(f"Google Drive token validation failed for user {user.id}: {str(e)}")
+        raise HTTPException(status_code=400, detail=f"Invalid Google Drive tokens: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)